src/rust/rust-proto/src/protocol/service_client.rs

```diff
@@ -17,12 +17,12 @@ pub trait Connectable {
 #[non_exhaustive]
 #[derive(Debug, thiserror::Error)]
 pub enum ConnectError {
-    #[error("failed to connect {0}")]
+    #[error("Failed to connect {0}")]
     ConnectionError(#[from] tonic::transport::Error),
 
-    #[error("healthcheck failed {0}")]
-    HealtcheckFailed(#[from] HealthcheckError),
+    #[error("Healthcheck failed for {0}: {1}")]
+    HealthcheckFailed(String, HealthcheckError),
```

Comment:
My preference here would be to do the following:
```
#[error("healthcheck failed for `{service_name}`: {source})]
HealthcheckFailed {
    service_name: String,
    source: HealthcheckError
}
```

This way we don't discard the Error's source, which could be useful information.

> The Error trait's source() method is implemented to return whichever field has a #[source] attribute or is named source, if any. This is for identifying the underlying lower level error that caused your error.
-- https://github.com/dtolnay/thiserror

Reply:
Can do